.pre-commit-config.yaml

```diff
@@ -20,11 +20,11 @@ repos:
     - id: typos
       exclude: _.*_dict.py
 - repo: https://github.com/astral-sh/ruff-pre-commit
-  # Ruff version.
-  rev: v0.5.1
+  rev: v0.9.1
```

Comment:
Why isn't this done automatically? Or don't we have that configured?

Reply:
Strange. Note that pre-commit updates are handled by pre-commit itself, not by GitHub Dependabot. I thought pre-commit updates itself by default, without the need for additional configuration.

In the meantime, I recommend we handle these updates manually. Because pre-commit has not been updated for a long time, substantial changes and bug fixes might be needed.

Reply:
It might be because that the two `repos` keys confuse pre-commit:
https://github.com/pydicom/pydicom/blob/54623aed1d43b73785c93976f96f4739f8473d71/.pre-commit-config.yaml#L3-L4
https://github.com/pydicom/pydicom/blob/54623aed1d43b73785c93976f96f4739f8473d71/.pre-commit-config.yaml#L16-L17

I'll fix that in a separate PR.

Reply:
See #2200.